dev_utils/src/utils.rs

```diff
@@ -2,21 +2,36 @@ use std::ops::{Add, Sub};
 
 use rand::{thread_rng, Rng};
 use rand_distr::Uniform;
+use num_traits::float::FloatCore;
 
 // random array that samples between min and max of T
 pub fn get_random_array<T>(n: usize, min_value: T, max_value: T) -> Vec<T>
-where
-    T: Copy + rand::distributions::uniform::SampleUniform,
+    where
+        T: Copy + rand::distributions::uniform::SampleUniform,
 {
     let rng = thread_rng();
     let uni = Uniform::new_inclusive(min_value, max_value);
     rng.sample_iter(uni).take(n).collect()
 }
 
+/// Inserts NaN values into the provided array at the specified frequency.
+/// * `name` - normalized_frequency - probability [0-1] of a NaN being inserted for each value.
+pub fn insert_nans<T>(values: &mut Vec<T>, normalized_frequency: f32)
+    where
+        T: FloatCore,
```

Comment:
This was done to be generic over all Float datatypes, which should help with benching/testing.
Not sure about fp16 support here though.

Reply:
Should be supported for f16 if we enable the `num-traits` feature
https://docs.rs/half/latest/half/struct.f16.html#impl-FloatCore-for-f16

Reply:
Can we do that? Won't that limit the potential targets as grows the dependency surface?

Reply:
On another note, any reason why you prefer [`FloatCore`](https://docs.rs/num-traits/0.2.14/num_traits/float/trait.FloatCore.html) over [`Float`](https://docs.rs/num-traits/0.2.14/num_traits/float/trait.Float.html)? I understand that the difference between both is that `FloatCore` works with [`no_std`](https://docs.rust-embedded.org/book/intro/no-std.html), but I have honestly no idea what this implies (& if this is important).. 

(P.S. I've been using `Float` in PR #21)

Reply:
Great question - I went with the `no_std` version because it runs on a greater number of systems and in smaller runtime configurations (think embedded systems, etc etc).

You can see an example of the platform targets here: https://doc.rust-lang.org/nightly/rustc/platform-support.html - in this case it's the `-none` targets that would be more suited.

The likelihood of someone using SIMD on those platforms is extremely limited, but *iff* there was no downside, FloatCore should be preferred based on my reading of the traits.

Reply:
Gotcha! Thanks for the clear explanation!

This got me wondering the following; this library uses (imports from) `std` quite a lot - it might be useful *(if there is a need for this)* to ease / possibly remove this dependency? Is certainly a very low prio TODO though :thinking: 

![image](https://user-images.githubusercontent.com/18898740/221205098-53a91276-bd85-4247-a16e-279cbd2d41a3.png)